lib/plop-pack/spec/helpers/title.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const hb = require('handlebars');
+const title = require('../../src/helpers/title');
+hb.registerHelper('title', title);
+
+const data = {
+  name: 'Fred',
+  age: 32,
+  favouriteIceCream: 'tuti fruti'
+};
+
+describe('handlebars template helper: title', () => {
+  it('should be a function', () => expect(title).toBeInstanceOf(Function));
+  it('should under-equals the given text', () => [
+      { text: 'A', result: 'A\n=' },
+      { text: 'Apples', result: 'Apples\n======' },
+      { text: 'A slightly longer example', result: 'A slightly longer example\n=========================' },
+      { text: '', result: '\n' }
+    ].forEach(test => expect(title({ fn: () => test.text })).toEqual(test.result)));
+  it('should produce a markdown title from a template and data', () => [
+      { text: '{{#title}}A{{/title}}', result: 'A\n=' },
+      { text: '{{#title}}Apples{{/title}}', result: 'Apples\n======' },
+      { text: '{{#title}}{{favouriteIceCream}}, yum!!{{/title}}', result: 'tuti fruti, yum!!\n=================' },
+      { text: '{{#title}}Profile: {{{name}}}, {{age}}{{/title}}', result: 'Profile: Fred, 32\n=================' },
+      { text: '{{#title}}{{{nothing}}}{{/title}}', result: '\n' }
+    ].forEach(test => expect(hb.compile(test.text)(data)).toEqual(test.result)));
```

Comment:
I don't think you need this.

Aren't you effectively just testing handlebars itself here?

Reply:
Though suppose you added it in response to my query, checking that it actually worked...

Reply:
I see it as a sort of integration level test.
The first lot checks the function works as expected.
The second lot checks I've got the contract of the function right by making sure it works in the handlebars context.

Reply:
The second lot basically saves having to actually manually check the result of plopping.
I mean you could add are ready that runs a plop and checks the title lines, but that feels like overkill.